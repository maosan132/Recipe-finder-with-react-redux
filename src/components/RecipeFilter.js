import React from 'react';
import PropTypes from 'prop-types';
import Form from 'react-bootstrap/Form';
import classes from './RecipeFilter.module.css';

const RecipeFilter = ({ handleFilter, recipes }) => {
  const categories = ['All']; // Start with the All category

  if (recipes) {
    recipes.map((r) => {
      if (!categories.includes(r.strCategory)) {
        categories.push(r.strCategory);
      }
      return categories;
    });
  }

  const handleClick = (e) => {
    const { value } = e.target;
    handleFilter(value);
    console.log(value);
  };

  return (
    <Form>
      <Form.Group className="d-flex justify-content-center p-2">
        <Form.Label htmlFor="select" className={`${classes.label} align-self-center`}>Select a category: </Form.Label>
        <Form.Control as="select" id="select" onChange={handleClick} className={`${classes.input} w-25`}>
          {categories.map((category) => (
            <option value={category} key={category}>
              {category}
            </option>
          ))}
        </Form.Control>
        <button type="button">Clear</button>
      </Form.Group>
    </Form>
  );
};

RecipeFilter.propTypes = {
  handleFilter: PropTypes.func.isRequired,
  recipes: PropTypes.arrayOf(PropTypes.shape({
    idMeal: PropTypes.string,
    strMeal: PropTypes.string,
    strCategory: PropTypes.string,
    strArea: PropTypes.string,
    strInstructions: PropTypes.string,
  })).isRequired,
};

export default RecipeFilter;

// import React from 'react';

// const RecipeFilter = () => (
//   <div>
//     A recipeFilter
//   </div>
// );

// export default RecipeFilter;
