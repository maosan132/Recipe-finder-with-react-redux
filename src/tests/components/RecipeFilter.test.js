import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { cleanup, render } from '@testing-library/react';
import '@testing-library/jest-dom';
import RecipeFilter from '../../components/RecipeFilter';

Enzyme.configure({ adapter: new Adapter() });

afterEach(cleanup);

const recipes = [
  { 0: { id: 1, name: 'Recipe1', category: 'Vegetarian' } },
  { 1: { id: 2, name: 'Recipe2', category: 'Seafood' } },
];

describe('<RecipeFilter />', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<RecipeFilter handleFilter={() => 'mock'} key={recipes.id} recipes={recipes} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  // it('it contains an option element', () => {
  //   const wrapper = shallow(
  //     <RecipeFilter
  //       handleFilter={() => 'mock'}
  //       key={recipes.id}
  //       recipes={recipes}
  //     />,
  //   );
  //   const option = <option>CATEGORIES</option>;
  //   expect(wrapper.containsMatchingElement(option)).toBe(true);
  // });

  // it('renders a select input', () => {
  //   const { queryAllByTestId } = render(
  //     <RecipeFilter handleFilter={() => 'mock'} key={recipes.id} recipes={recipes} />,
  //   );
  //   const result = queryAllByTestId('categories');
  //   expect(result).toBeTruthy();
  //   expect(result[0]).toHaveTextContent('CATEGORIES');
  //   expect(result[0]).toHaveTextContent('All');
  // });
});
