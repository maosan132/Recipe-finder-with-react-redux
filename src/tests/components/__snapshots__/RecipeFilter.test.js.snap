// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecipeFilter /> renders correctly 1`] = `
<form
  className=""
>
  <div
    className="d-flex justify-content-center p-2"
  >
    <label
      className="label align-self-center form-label"
      htmlFor="select"
    >
      Select a category: 
    </label>
    <select
      className="input w-25 form-control"
      id="select"
      onChange={[Function]}
    >
      <option
        value="All"
      >
        All
      </option>
      <option />
    </select>
    <button
      onClick={[Function]}
      type="button"
    >
      Clear
    </button>
  </div>
</form>
`;
